### PR TITLE
Support stateful on displaced interface

### PR DESCRIPTION
**Pull request by lindsayad**
Previously if there was a displaced material and interface objects and not boundary objects on the same boundary as the interface object, you could get bad computations on the displaced material because the displaced assembly would not be properly initialized (nor likely the displaced systems)

Refs #21900

### CODE REVIEW

**Review by GiudGiud — CHANGES_REQUESTED** [review]
needs a test

File: framework/include/loops/ThreadedElementLoopBase.h [hunk 1]
```diff
@@ -151,6 +151,11 @@ class ThreadedElementLoopBase
   virtual bool keepGoing() { return true; }
 
 protected:
+  /**
+   * emit a relatively error message when we catch a MetaPhysicL logic error
```
Comment (GiudGiud) [inline framework/include/loops/ThreadedElementLoopBase.h:155]:
```suggestion
   * emit a relatively clear error message when we catch a MetaPhysicL logic error
```

File: framework/include/loops/ThreadedElementLoopBase.h [hunk 2]
```diff
@@ -193,6 +198,19 @@ ThreadedElementLoopBase<RangeType>::~ThreadedElementLoopBase()
...
+template <typename RangeType>
+void
+ThreadedElementLoopBase<RangeType>::translateMetaPhysicLError(const MetaPhysicL::LogicError &)
+{
+  mooseError(
+      "We caught a MetaPhysicL error in while performing element loops. This is potentially "
+      "due to AD not having a sufficiently large derivative container size. To increase the AD "
+      "container size, you can run configure in the MOOSE root directory with the "
+      "'--with-derivative-size=<n>' option and then recompile. Other causes of MetaPhysicL logic "
+      "errors include evaluating functions like sqrt (which gets called for vector norm functions) "
```
Comment (GiudGiud) [inline framework/include/loops/ThreadedElementLoopBase.h:210]:
```suggestion
      "errors include evaluating functions where they are not defined or differentiable like sqrt (which gets called for vector norm functions) "
```

File: modules/fsi/test/tests/2d-finite-strain-steady/2layers_2d_midline.msh [hunk 3]
```diff
@@ -0,0 +1,283 @@
+$MeshFormat
```
Comment (GiudGiud) [inline modules/fsi/test/tests/2d-finite-strain-steady/2layers_2d_midline.msh:1]: document the gmsh version somewhere pls

**Review by GiudGiud — CHANGES_REQUESTED** [review]

File: modules/fsi/test/tests/2d-finite-strain-steady/2layers_2d_midline.msh (hunk 3 above)
Comment (lindsayad) [reply modules/fsi/test/tests/2d-finite-strain-steady/2layers_2d_midline.msh:1]: It's documented right in the line below

File: framework/include/loops/ThreadedElementLoopBase.h (hunk 1 above)
Comment (GiudGiud) [inline framework/include/loops/ThreadedElementLoopBase.h:155]:
```suggestion
   * emit a relatively clear error message when we catch a MetaPhysicL logic error
```

File: modules/fsi/test/tests/2d-finite-strain-steady/2layers_2d_midline.msh (hunk 3 above)
Comment (GiudGiud) [reply modules/fsi/test/tests/2d-finite-strain-steady/2layers_2d_midline.msh:1]: nice

**Review by GiudGiud — APPROVED** [review]

**Comment by lindsayad** [comment]
Thanks for the fast reviews!

**Pull request merged**